src/chunk_insert_state.c

```diff
@@ -204,6 +205,168 @@ create_chunk_result_relation_info(ChunkDispatch *dispatch, Relation rel, Index r
+											 int map_size, Index varno, Oid rowtype,
+											 TupleDesc chunk_desc)
+{
+	bool		found_whole_row;
+	int			i;
+	ListCell   *lc,
+			   *prev,
+			   *next;
+
+	Assert(update_tles != NIL);
+	/* map hypertable attnos -> chunk attnos for the hypertable */
+	update_tles = (List *) map_variable_attnos_compat((Node *) update_tles, varno, 0, map, map_size,
+													  rowtype, &found_whole_row);
+	/* map hypertable attnos -> chunk attnos for the "excluded" table */
+	update_tles = (List *) map_variable_attnos_compat((Node *) update_tles, INNER_VAR, 0, map, map_size,
+													  rowtype, &found_whole_row);
+
+
+	Assert(list_length(update_tles) == map_size);
+
+	/*
+	 * delete any hypertable tle entries for which there are no chunk
+	 * attributes. This can happen for dropped columns that exist on the
+	 * hyperable but not on the chunks.
+	 */
+	lc = list_head(update_tles);
+	prev = NULL;
+	for (i = 0; i < map_size; i++)
+	{
+		AttrNumber	att = map[i];
+
+		next = lnext(lc);
+
+		if (att == InvalidAttrNumber)
+			update_tles = list_delete_cell(update_tles, lc, prev);
+		else
+			prev = lc;
+		lc = next;
+	}
+
+	/*
+	 * Double-check that the TLE is in the same order as the chunk attributes
+	 * and also fix up the TLE resno after the deletes above
+	 */
+	i = 0;
+	foreach(lc, update_tles)
+	{
+		TargetEntry *t = lfirst(lc);
+		Form_pg_attribute attribute;
+
+		Assert(i < chunk_desc->natts);
+		attribute = chunk_desc->attrs[i];
+
+		if (strcmp(NameStr(attribute->attname), t->resname) != 0)
+			elog(ERROR, "TLE order in onConflictUpdate does not match chunk attribute order");
+
+		t->resno = i + 1;
+		i++;
+	}
+
+	/* set the projection result slot to the chunk schema */
+	ExecSetSlotDescriptor(get_projection_info_slot_compat(orig), chunk_desc);
+	return ExecBuildProjectionInfoCompat(update_tles, orig->pi_exprContext,
+										 get_projection_info_slot_compat(orig), NULL, chunk_desc);
+}
+
+/* Change the projections to work with chunks instead of hypertables */
+static void
+adjust_projections(ChunkInsertState *cis, ChunkDispatch *dispatch, Oid rowtype)
+{
+	ResultRelInfo *rri = cis->result_relation_info;
+	AttrNumber *variable_attnos_map;
+	int			variable_attnos_map_size;
+	TupleDesc	chunk_desc = cis->tup_conv_map->outdesc;
+	TupleDesc	hypertable_desc = cis->tup_conv_map->indesc;
+
+	Assert(cis->tup_conv_map != NULL);
+
+	/*
+	 * we need the opposite map from cis->tup_conv_map. The map needs to have
+	 * the hypertable_desc in the out spot for map_variable_attnos to work
+	 * correctly in mapping hypertable attnos->chunk attnos
+	 */
+	variable_attnos_map = convert_tuples_by_name_map(chunk_desc,
+													 hypertable_desc,
+													 gettext_noop("could not convert row type"));
+	variable_attnos_map_size = hypertable_desc->natts;
+
+	if (rri->ri_projectReturning != NULL)
+	{
+		rri->ri_projectReturning =
+			get_adjusted_projection_info_returning(
```

Comment:
Inconsistent use of initial newline after opening paranthesis.
Also, for consistency, put one arg per line if breaking a function like this (also applies below)